io/src/obj_io.cpp

```diff
@@ -1108,14 +1117,14 @@ pcl::io::saveOBJFile (const std::string &file_name,
     {
       // Write faces with "f"
       fs << "f";
-      // There's one UV per vertex per face, i.e., the same vertex can have
-      // different UV depending on the face.
       for (std::size_t j = 0; j < tex_mesh.tex_polygons[m][i].vertices.size (); ++j)
       {
         std::uint32_t idx = tex_mesh.tex_polygons[m][i].vertices[j] + 1;
-        fs << " " << idx
-           << "/" << tex_mesh.tex_polygons[m][i].vertices.size () * (i+f_idx) +j+1
-           << "/" << idx; // vertex index in obj file format starting with 1
+        fs << " " << idx << "/";
+        // texture coordinate indices are optional
+        if (!tex_mesh.tex_coord_indices[m][i].vertices.empty())
```

Comment:
If this condition is false, the slash would be printed in L1123 and 1127

Reply:
Yes, that is intentional. Vertices of a face are specified as `a/b/c`, where `a` is the vertex/point index, `b` is the texture coordinate index, and `c` is the normal index. `b` and `c` are optional. If `b` is left out, the correct form is `a//b` (with two slashes). See also the Wikipedia page I linked above